src/common/pch.h

```diff
@@ -19,8 +19,10 @@
 #define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_TRACE
 #include <spdlog/spdlog.h>
 
-// Boost Assert
+// Boost libraries
+#include <boost/algorithm/string/trim.hpp>
 #include <boost/assert.hpp>
+#include <boost/lexical_cast.hpp>
 #include <boost/static_assert.hpp>
```

Comment:
Unsure if we should add includes used in only a few places in this file. Is there really an advantage to have all those uncommon includes here?

Reply:
> Is there really an advantage to have all those uncommon includes here?

Yes and no, 

On one side we can take advantage that the precompiled header file will pre-parse and precompiled all of those files, 
speeding up compilation by at tad (since we would probably never change the source of the boost libraries.)
 
On the other hand, it can make it just a bit more confusing 🤏 to know were certain functions came from.

Since `boost::trim` and `boost::lexical_cast` was only ever used in the testing sensors it wouldn't make much difference if it was included directly there, 
but if someone were to change those files or use them somewhere else, they wouldn't have to parse `<boost/algorithm/string/trim.hpp>` and `<boost/lexical_cast.hpp>` again. 

But you are right, since its quite a small file the compilation time gains would be minimal/negligible.

**Do you want me to move them back to only `TestingSensors`?**

Reply:
Yeah, let's keep `pch.h` to only header files that are found almost everywhere. From the clang docs (https://clang.llvm.org/docs/PCHInternals.html):
```
The use case for precompiled headers is relatively simple: when there is a common set of headers that is included in nearly every source file in the project, we precompile that bundle of headers into a single precompiled header (PCH file).
```